core/files/uploads-manager.php

```diff
@@ -628,6 +628,64 @@ private function validate_file( array $file, $file_extensions = [] ) {
 		return $file_type_handler->validate_file( $file );
 	}
 
+	/**
+	 * Checks whether the provided path is an absolute path.
+	 *
+	 * @since 3.20.0
+	 * @access private
+	 *
+	 * @param string $path
+	 * @return bool
+	 */
+	private function is_absolute_path( $path ) {
+		$realpath = realpath( $path );
+
+		if ( false === $realpath ) {
+			$path_parts = explode( DIRECTORY_SEPARATOR, $path );
+
+			foreach ( $path_parts as $part ) {
+				if ( in_array( [ '.', '..' ], $part ) ) {
+					return false;
+				}
+			}
+		}
+
+		if ( DIRECTORY_SEPARATOR === $realpath[0] ?? preg_match( '/^[a-zA-Z]:[\/\\\\]/', $path ) ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks whether the provided path is a temporary file.
+	 *
+	 * @since 3.20.0
+	 * @access private
+	 *
+	 * @param string $path
+	 * @return bool
+	 */
+	private function is_temporary_file( $path ) {
+		$realpath = is_executable( dirname( $path ) ) ? realpath( $path ) : $path;
+
+		if ( false === $realpath ) {
+			return false;
+		}
+
+		if ( is_uploaded_file( $path ) ) {
+			return true;
+		}
+
+		foreach ( $this->temp_unique_dirs as $temp_dir ) {
+			if ( strpos( $realpath, $temp_dir ) === 0 ) {
+				return true;
+			}
+		}
```

Comment:
same as above
this is not always right, for example in cases where the path looks like this:

```PHP
$uploads_dir = '/tmp/uploads';
$path = '/tmp/uploads/../../var/www/html';
```